### PR TITLE
fix: init pre_authorization does not allow negative timestamps as input

### PR DESCRIPTION
**Pull request by dcaf-mocha**

### CODE REVIEW

File: programs/pre-authorized-debit-v1/src/errors.rs [hunk 1]
```diff
@@ -44,4 +44,7 @@ pub enum CustomProgramError {
 
     #[msg("Only token account owner can pause a pre-authorization")]
     PausePreAuthorizationUnauthorized, // 6013 or 0x177D
+
+    #[msg("Invalid timestamp value provided")]
+    InvalidTimestamp, // 6014 or 0x177E
```
Comment (dcaf-mocha) [inline programs/pre-authorized-debit-v1/src/errors.rs:49]: I dont think this will ever get called. Rust will just try and parse the bytes as u64 if a client tries passing a negative value.

**Review by dcaf-matcha — CHANGES_REQUESTED** [review]
Can you add a test where you send in u64 max and see what happen? I wonder if it'll turn that into a huge negative number coz of 2's complement or just panic

**Review by dcaf-matcha — CHANGES_REQUESTED** [review]
After thinking through it again, I think we can abandon most of the changes in this PR and just support negative timestamps. There are a few semi-related changes to make also. I'll itemize them along with the main steps here:

1. Revert everything to what it is on main
2. All timestamps on main afaik are already passed in and stored as i64 types which is valid. The only time-based parameter that isn't passed in or stored as i64 is `repeat_frequency_seconds` for a recurring pre-authorization. I think this is okay given that has to be positive always but to avoid weird size issues, we should add a validation in `init_pre_authorization` that checks that this value is < i64::MAX as u64.
3. Even though the actual values, by the time the runtime gets to compute_current_cycle will never satisfy current_timestamp < activation_timestamp due to prior checks, we should still assert this and error out if that is the case to avoid weird bugs that can be introduced in future iterations when the original check is removed. I'd also unit test this logic.
4. Add integration test cases for changes in point 2 above. Maybe also add a test case where you set the timestamps to negative and things still function as expected.

**Comment by dcaf-mocha** [comment]
Closing in place of https://github.com/dcaf-labs/pre-authorized-debit/pull/31
